<a href="https://colab.research.google.com/github/hansjang/Py_Radar/blob/main/%EB%B9%85%EB%B6%84%EA%B8%B0Part5_Exam7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!git clone https://github.com/AnalyticsKnight/yemoonsaBigdata

Cloning into 'yemoonsaBigdata'...
remote: Enumerating objects: 709, done.
remote: Counting objects: 100% (346/346), done.
remote: Compressing objects: 100% (231/231), done.
remote: Total 709 (delta 156), reused 237 (delta 115), pack-reused 363 (from 1)
Receiving objects: 100% (709/709), 11.46 MiB | 7.44 MiB/s, done.
Resolving deltas: 100% (341/341), done.
Updating files: 100% (173/173), done.


# 1. 작업형 제1유형

## 01. 학생점수

In [20]:
import pandas as pd
import numpy as np
from tabulate import tabulate

df = pd.read_csv('/content/yemoonsaBigdata/datasets/Part5/ex7/07.01.01-students_scores.csv')

print(df.info())

# 과목별 수강 카운트
df1 = df[['Statistics', 'DataScience', 'MachineLearning', 'EnglishLiterature']]

subject_counts = df1.count()

sorted_subject_counts = subject_counts.sort_values(ascending=False)

print(sorted_subject_counts)

print(tabulate(sorted_subject_counts.items(), headers=['Subject', 'Count'], tablefmt='grid'))

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   student id         1000 non-null   int64  
 1   Statistics         773 non-null    float64
 2   DataScience        709 non-null    float64
 3   MachineLearning    769 non-null    float64
 4   EnglishLiterature  749 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 39.2 KB
None
Statistics           773
MachineLearning      769
EnglishLiterature    749
DataScience          709
dtype: int64
+-------------------+---------+
| Subject           |   Count |
+===================+=========+
| Statistics        |     773 |
+-------------------+---------+
| MachineLearning   |     769 |
+-------------------+---------+
| EnglishLiterature |     749 |
+-------------------+---------+
| DataScience       |     709 |
+-------------------+---------+


**가장 많은 학생이 들었던 과목 : 통계(Statistics)**


In [24]:
# 결측치 제거
df1_stat = df1['Statistics'].dropna()
print(df1_stat.info())

# 과목점수 표준화
import scipy.stats as stats
df1_stat_score = stats.zscore(df1_stat)
print(df1_stat_score)

# 최대값 출력
result = round(max(df1_stat_score), 2)
print(result)


<class 'pandas.core.series.Series'>
Index: 773 entries, 1 to 998
Series name: Statistics
Non-Null Count  Dtype  
--------------  -----  
773 non-null    float64
dtypes: float64(1)
memory usage: 12.1 KB
None
1      0.928049
2     -1.537180
3      1.209789
5     -1.114570
6     -0.832829
         ...   
994    0.540656
995   -0.586306
996    0.012392
997    0.963266
998   -0.234131
Name: Statistics, Length: 773, dtype: float64
1.74


**표준점수 최대값 : 1.74**

##02. 물고기 : 종속변수 > 무게(Weight)와 가장 높은 상관관계를 보이는 변수

In [30]:
df = pd.read_csv('/content/yemoonsaBigdata/datasets/Part5/ex7/07.01.02-fish_weight_data.csv')

# print(df.info())

# ID 변수 제거하기
df1 = df.drop('id', axis=1)

# 상관계수 구하기
corr_matrix = df1.corr()
# print(corr_matrix)

# 종속변수(Weight) 기준으로 상관계수 구하기
corr_weight = corr_matrix['Weight'].drop('Weight')
print(corr_weight)

Length     0.947420
Height     0.222426
Width     -0.046809
FinSize    0.107433
Name: Weight, dtype: float64


**무게(Weight)와 가장 높은 상관관계를 보이는 변수 : 길이(Length)**



## 03. 아파트 가격 : 이상치가 가장 많은 독립변수의 이상치 개수

In [37]:
df = pd.read_csv('/content/yemoonsaBigdata/datasets/Part5/ex7/07.01.03-apartment_prices_dataset.csv')

# print(df.info())
# print(df.describe())

# 결측치 제거
df1 = df.dropna()
print(df1.info())

# 이상치 구하기
cols = df1.columns[:-1]
o_cnt_per_var = {}

for c in cols:
  Q1, Q3 = df1[c].quantile([0.25, 0.75])
  IQR = Q3 - Q1
  o_cnt = ((df1[c] < (Q1-1.5*IQR)) | (df1[c] > (Q3+1.5*IQR))).sum()

  o_cnt_per_var[c] = o_cnt

print(o_cnt_per_var)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 735 entries, 0 to 734
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Size              735 non-null    float64
 1   Bedrooms          735 non-null    int64  
 2   Bathrooms         735 non-null    int64  
 3   Floor             735 non-null    int64  
 4   Age               735 non-null    int64  
 5   DistanceToCenter  735 non-null    float64
 6   NearbySchools     735 non-null    float64
 7   NearbyParks       735 non-null    float64
 8   PublicTransport   735 non-null    float64
 9   SafetyIndex       735 non-null    int64  
 10  Price             735 non-null    float64
dtypes: float64(6), int64(5)
memory usage: 63.3 KB
None
{'Size': 1, 'Bedrooms': 3, 'Bathrooms': 2, 'Floor': 4, 'Age': 15, 'DistanceToCenter': 1, 'NearbySchools': 5, 'NearbyParks': 4, 'PublicTransport': 5, 'SafetyIndex': 3}


In [40]:
# 이상치가 가장 많은 변수 구하기
result = max(o_cnt_per_var, key=o_cnt_per_var.get) # get key with max value
print(result)

이상치가 가장 많은 변수 : Age


**이상치가 가장 많은 변수 : Age**


# 2. 작업형 제2유형

## 02. 신규 영업지점 설립 : 매출 예측 RMSE

In [45]:
# (1) 사전 준비
## 필요한 module import
import pandas as pd
from tabulate import tabulate

## 데이터 불러오기
train = pd.read_csv('/content/yemoonsaBigdata/datasets/Part5/ex7/07.02.01-sales_train_dataset.csv')
X_test = pd.read_csv('/content/yemoonsaBigdata/datasets/Part5/ex7/07.02.02-sales_test_dataset_x.csv')
y_test = pd.read_csv('/content/yemoonsaBigdata/datasets/Part5/ex7/07.02.03-sales_test_dataset_y.csv')

## 데이터 살펴보기
print(tabulate(train.head(), headers='keys', tablefmt='grid'))
print(tabulate(X_test.head(), headers='keys', tablefmt='grid'))
print(tabulate(y_test.head(), headers='keys', tablefmt='grid'))

+----+--------------+--------+--------------+----------------------------+-----------------+----------------+-----------+
|    | BranchName   | City   |   Population |   IncomeGeneratingPopRatio |   AverageIncome | IndustryType   |     Sales |
+====+==============+========+==============+============================+=================+================+===========+
|  0 | 지점106      | 대전   |      1961994 |                   0.201855 |         5104676 | 농업           | 256900147 |
+----+--------------+--------+--------------+----------------------------+-----------------+----------------+-----------+
|  1 | 지점69       | 수원   |      1999348 |                   0.609039 |         6175887 | 서비스업       | 185564838 |
+----+--------------+--------+--------------+----------------------------+-----------------+----------------+-----------+
|  2 | 지점480      | 광주   |      1736191 |                   0.684106 |         4935993 | 농업           | 265883549 |
+----+--------------+--------+--------------

In [46]:
# (2) 데이터 전처리
## 데이터 타입 확인
print(train.info())
print(X_test.info())
print(y_test.info())

## 결측치 확인
print(train.isnull().sum())
print(X_test.isnull().sum())
print(y_test.isnull().sum())

## 변수 타입별 구분
COL_DEL = ['BranchName']  #불필요한 변수
COL_NUM = ['Population', 'IncomeGeneratingPopRatio', 'AverageIncome']   #수치형변수
COL_CAT = ['City', 'IndustryType']  #범주형변수
COL_Y = ['Sales'] #종속변수

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 7 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   BranchName                700 non-null    object 
 1   City                      700 non-null    object 
 2   Population                700 non-null    int64  
 3   IncomeGeneratingPopRatio  700 non-null    float64
 4   AverageIncome             700 non-null    int64  
 5   IndustryType              700 non-null    object 
 6   Sales                     700 non-null    int64  
dtypes: float64(1), int64(3), object(3)
memory usage: 38.4+ KB
None
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 6 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   BranchName                300 non-null    object 
 1   City                      300 non-null    object 
 2   Popu

In [47]:
# (3) 모형 구축
## train 데이터 독립변수, 종속변수 세트 분리
X_train = train[COL_CAT + COL_NUM]
y_train = train[COL_Y]

# print(tabulate(X_train.head(), headers='keys', tablefmt='grid'))
# print(tabulate(y_train.head(), headers='keys', tablefmt='grid'))

## 데이터 분할 => 훈련(7): 검증(3)
from sklearn.model_selection import train_test_split
X_tr, X_val, y_tr, y_val = train_test_split(X_train,
                                            y_train.values.ravel(), #y_train을 1차원 Numpy 배열(array)로 평탄화하여 반환
                                            test_size = 0.3)

print(tabulate(X_tr.head(), headers='keys', tablefmt='grid'))
# print(tabulate([[val] for val in y_tr[:5]], headers=["y_tr"], tablefmt='grid'))
# print(y_tr[:5])

## 수치형 변수 표준화
## code 2번 돌면 작살나는 구간 ##
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()
scaler.fit(X_tr[COL_NUM])

X_tr[COL_NUM] = scaler.transform(X_tr[COL_NUM])
X_val[COL_NUM] = scaler.transform(X_val[COL_NUM])
X_test[COL_NUM] = scaler.transform(X_test[COL_NUM])

print(tabulate(X_tr.head(), headers='keys', tablefmt='grid'))

## 범주형 변수 인코딩
## code 2번 돌면 작살나는 구간 ##
from sklearn.preprocessing import LabelEncoder

X = pd.concat([X_train[COL_CAT], X_test[COL_CAT]])

for col in COL_CAT:
  le = LabelEncoder()
  le.fit(X[col])
  X_tr[col] = le.transform(X_tr[col])
  X_val[col] = le.transform(X_val[col])
  X_test[col] = le.transform(X_test[col])

  print(col)
  print(le.classes_)
  print('\n')

+-----+--------+----------------+--------------+----------------------------+-----------------+
|     | City   | IndustryType   |   Population |   IncomeGeneratingPopRatio |   AverageIncome |
+=====+========+================+==============+============================+=================+
| 312 | 용인   | 제조업         |      1554172 |                   0.400235 |         4490972 |
+-----+--------+----------------+--------------+----------------------------+-----------------+
| 541 | 부산   | 제조업         |      1040054 |                   0.564189 |         6140897 |
+-----+--------+----------------+--------------+----------------------------+-----------------+
| 378 | 부산   | 농업           |      1713513 |                   0.500972 |         6975107 |
+-----+--------+----------------+--------------+----------------------------+-----------------+
| 342 | 인천   | 제조업         |      2706807 |                   0.50359  |         5649170 |
+-----+--------+----------------+--------------+-----------

In [48]:
print(tabulate(X_tr.head(), headers='keys', tablefmt='grid'))

+-----+--------+----------------+--------------+----------------------------+-----------------+
|     |   City |   IndustryType |   Population |   IncomeGeneratingPopRatio |   AverageIncome |
+=====+========+================+==============+============================+=================+
| 312 |      7 |              3 |   -0.288066  |                  -0.349269 |       -0.458227 |
+-----+--------+----------------+--------------+----------------------------+-----------------+
| 541 |      4 |              3 |   -1.00438   |                   0.777819 |        0.962022 |
+-----+--------+----------------+--------------+----------------------------+-----------------+
| 378 |      4 |              0 |   -0.0660573 |                   0.343237 |        1.68011  |
+-----+--------+----------------+--------------+----------------------------+-----------------+
| 342 |      9 |              3 |    1.31789   |                   0.361237 |        0.538745 |
+-----+--------+----------------+-------

In [51]:
## 1) 랜덤포레스트
from sklearn.ensemble import RandomForestRegressor
modelRF = RandomForestRegressor(random_state = 123)
modelRF.fit(X_tr, y_tr)

## 2) XGBoost
from xgboost import XGBRegressor
modelXGB = XGBRegressor(objective='reg:squarederror', random_state = 123)
modelXGB.fit(X_tr, y_tr)

XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, device=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=None, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             multi_strategy=None, n_estimators=None, n_jobs=None,
             num_parallel_tree=None, random_state=123, ...)

In [52]:
# (4) 모델 평가
y_val_predRF = modelRF.predict(X_val)
y_val_predXGB = modelXGB.predict(X_val)

In [55]:
## 답안 채점 기준인 RMSE 계산(MSE를 사용해도 됨)
from sklearn.metrics import mean_squared_error
import numpy as np

def cal_rmse(actual, pred):
  return np.sqrt(mean_squared_error(actual, pred))

scoreRF = cal_rmse(y_val, y_val_predRF)
scoreXGB = cal_rmse(y_val, y_val_predXGB)

print('Random Forest: \t', scoreRF)
print('XGBoost: \t', scoreXGB)

Random Forest: 	 123094328.01620951
XGBoost: 	 139191781.57135928


In [60]:
## 하이퍼파라미터 튜닝
modelRF2 =  RandomForestRegressor(
                          n_estimators = 50,
                          max_depth = 5,
                          random_state = 123)
modelRF2.fit(X_tr, y_tr)
y_val_predRF2 = modelRF2.predict(X_val)
scoreRF2 = cal_rmse(y_val, y_val_predRF2)
print('Random Forest 2: \t', scoreRF2)

modelRF3 =  RandomForestRegressor(
                          n_estimators = 50,
                          max_depth = 10,   # 5 -> 10
                          random_state = 123)
modelRF3.fit(X_tr, y_tr)
y_val_predRF3 = modelRF3.predict(X_val)
scoreRF3 = cal_rmse(y_val, y_val_predRF3)
print('Random Forest 3: \t', scoreRF3)

modelRF4 =  RandomForestRegressor(
                          n_estimators = 100, # 50 -> 100
                          max_depth = 10,
                          random_state = 123)
modelRF4.fit(X_tr, y_tr)
y_val_predRF4 = modelRF4.predict(X_val)
scoreRF4 = cal_rmse(y_val, y_val_predRF4)
print('Random Forest 4: \t', scoreRF4)

Random Forest 2: 	 118299482.38343054
Random Forest 3: 	 121608482.04235359
Random Forest 4: 	 120446145.48458475


In [61]:
# (5) RMSE값이 가장 낮은 모형 최종 선택 후 제출
pred = modelRF2.predict(X_test[COL_CAT+COL_NUM])
result = pd.DataFrame({'BranchNAME': X_test.BranchName,
                       'Sales': pred})
print(tabulate(result.head(), headers='keys', tablefmt='grid'))

+----+--------------+-------------+
|    | BranchNAME   |       Sales |
+====+==============+=============+
|  0 | 지점994      | 2.92676e+08 |
+----+--------------+-------------+
|  1 | 지점860      | 3.61006e+08 |
+----+--------------+-------------+
|  2 | 지점299      | 3.06466e+08 |
+----+--------------+-------------+
|  3 | 지점554      | 3.14238e+08 |
+----+--------------+-------------+
|  4 | 지점673      | 3.04766e+08 |
+----+--------------+-------------+


In [65]:
# to_csv 함수로 제출
result.to_csv('빅분기Part5_Exam7_작업형제2유형_결과.csv', index=False, encoding='utf-8-sig')

# 3. 작업형 제3유형

## 01. 로지스틱 회귀모형 : 오즈비, 로짓우도, 오차율

In [2]:
# library import
import pandas as pd
import numpy as np
from tabulate import tabulate
import statsmodels.api as sm

df = pd.read_csv('/content/yemoonsaBigdata/datasets/Part5/ex7/07.03.01-gender_prediction_dataset.csv')

print(tabulate(df.head(), headers='keys', tablefmt='grid' ))

+----+----------+----------+------------+----------+
|    |   Height |   Weight |   ShoeSize | Gender   |
+====+==========+==========+============+==========+
|  0 |    177.6 |     59   |        274 | Male     |
+----+----------+----------+------------+----------+
|  1 |    168   |     49.4 |        238 | Female   |
+----+----------+----------+------------+----------+
|  2 |    157.4 |     69.8 |        248 | Female   |
+----+----------+----------+------------+----------+
|  3 |    177.9 |     62.3 |        278 | Male     |
+----+----------+----------+------------+----------+
|  4 |    156   |     55.3 |        232 | Female   |
+----+----------+----------+------------+----------+


In [4]:
# Gender를 이진변수로 변환하기
print(df['Gender'].unique())

df1 = df.copy()
df1['Gender'] = df1['Gender'].map({'Male':1, 'Female':0})
print(df1['Gender'].unique())

['Male' 'Female']
[1 0]


In [7]:
# 변수 설정하기
X = df1[['Height', 'Weight', 'ShoeSize']]
y = df1['Gender']

# 데이터 분리하기
from sklearn.model_selection import train_test_split

# stratify 옵션 : y 클래스의 비율을 유지하기 위해 사용
X_train, X_test, y_train, y_test = train_test_split(X, y.values.ravel(), test_size = 0.3, stratify=y.values.ravel())


print(len(X_train), len(X_test))

210 90


In [9]:
# 로지스틱 회귀 모형 훈련 생성 및 훈련
X_train = sm.add_constant(X_train)  # 절편 추가
lm = sm.Logit(y_train, X_train.values).fit()  # X_train을 ndarray로 입력

# 모형 요약
print(lm.summary())

Optimization terminated successfully.
         Current function value: 0.048987
         Iterations 11
                           Logit Regression Results                           
Dep. Variable:                      y   No. Observations:                  210
Model:                          Logit   Df Residuals:                      206
Method:                           MLE   Df Model:                            3
Date:                Sat, 24 Aug 2024   Pseudo R-squ.:                  0.9293
Time:                        01:00:18   Log-Likelihood:                -10.287
converged:                       True   LL-Null:                       -145.55
Covariance Type:            nonrobust   LLR p-value:                 2.372e-58
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const       -152.0114     39.484     -3.850      0.000    -229.398     -74.625
x1             0.4450      0

In [13]:
# odds ratio (오즈비) 구하기
# 오즈는 성공할 확률과 실패할 확률의 비이다
# np.exp는 지수를 얻기 위한 것으로 회귀계수(coef)를 변환하면 오즈비가 된다
odds_ratios = np.exp(lm.params)
print(odds_ratios)
result = odds_ratios[2]
print(result)

[9.60081769e-67 1.56043186e+00 1.42843621e+00 1.24591193e+00]
1.4284362094473393


In [15]:
# 로그 우도 구하기
log_likelihood = lm.llf
print(log_likelihood)

-10.287199951476525


In [19]:
# 평가용 데이터로 예측하기
X_test = sm.add_constant(X_test) # 절편 추가
y_pred = (lm.predict(X_test) >= 0.5).astype(int)

# error를 구해서 출력하기
error_rate = round(np.mean(np.abs(y_pred.values - y_test)), 4)
print(error_rate)

0.0111


## 02. 다중회귀분석 : 결정계수, 회귀계수, p-value

In [20]:
# library import
import pandas as pd
import statsmodels.api as sm
from tabulate import tabulate

df = pd.read_csv('/content/yemoonsaBigdata/datasets/Part5/ex7/07.03.02-used_car_price_dataset.csv')

print(tabulate(df.head(), headers='keys', tablefmt='grid'))

+----+--------+---------+----------------+-----------+------------+-------+-------+--------------+
|    |   year |   price | transmission   |   mileage | fuelType   |   tax |   mpg |   engineSize |
+====+========+=========+================+===========+============+=======+=======+==============+
|  0 |   2017 |   12500 | Manual         |     15735 | Petrol     |   150 |  55.4 |          1.4 |
+----+--------+---------+----------------+-----------+------------+-------+-------+--------------+
|  1 |   2016 |   16500 | Automatic      |     36203 | Diesel     |    20 |  64.2 |          2   |
+----+--------+---------+----------------+-----------+------------+-------+-------+--------------+
|  2 |   2016 |   11000 | Manual         |     29946 | Petrol     |    30 |  55.4 |          1.4 |
+----+--------+---------+----------------+-----------+------------+-------+-------+--------------+
|  3 |   2017 |   16800 | Automatic      |     25952 | Diesel     |   145 |  67.3 |          2   |
+----+----

In [23]:
# 라벨인코딩
print(df['transmission'].unique())
df['transmission'] = df['transmission'].map({'Manual': 0, 'Automatic' : 1, 'Semi-Auto' : 2 })
print(df['fuelType'].unique())
df['fuelType'] = df['fuelType'].map({'Petrol' : 0, 'Diesel' : 1, 'Hybrid' : 2})

print(tabulate(df.head(), headers='keys', tablefmt='grid'))

['Manual' 'Automatic' 'Semi-Auto']
['Petrol' 'Diesel' 'Hybrid']
+----+--------+---------+----------------+-----------+------------+-------+-------+--------------+
|    |   year |   price |   transmission |   mileage |   fuelType |   tax |   mpg |   engineSize |
+====+========+=========+================+===========+============+=======+=======+==============+
|  0 |   2017 |   12500 |              0 |     15735 |          0 |   150 |  55.4 |          1.4 |
+----+--------+---------+----------------+-----------+------------+-------+-------+--------------+
|  1 |   2016 |   16500 |              1 |     36203 |          1 |    20 |  64.2 |          2   |
+----+--------+---------+----------------+-----------+------------+-------+-------+--------------+
|  2 |   2016 |   11000 |              0 |     29946 |          0 |    30 |  55.4 |          1.4 |
+----+--------+---------+----------------+-----------+------------+-------+-------+--------------+
|  3 |   2017 |   16800 |              1 |   

In [24]:
# 독립변수, 종속변수 설정
X_cols = ['year', 'transmission', 'mileage', 'fuelType', 'tax', 'mpg', 'engineSize']
y_col = ['price']

# 독립변수, 종속변수 데이터 분리
X = df[X_cols]
y = df[y_col]

In [26]:
# 다중회귀분석 수행
X = sm.add_constant(X)  # 절편(상수항) 추가
model = sm.OLS(y, X).fit()  # 분석 수행 - 모델 생성

# 결과 요약
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                  price   R-squared:                       0.796
Model:                            OLS   Adj. R-squared:                  0.796
Method:                 Least Squares   F-statistic:                     2784.
Date:                Sat, 24 Aug 2024   Prob (F-statistic):               0.00
Time:                        01:35:51   Log-Likelihood:                -49990.
No. Observations:                5000   AIC:                         1.000e+05
Df Residuals:                    4992   BIC:                         1.000e+05
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const        -4.797e+06   1.46e+05    -32.746   

In [27]:
# 결정계수 구하기
r_squared = model.rsquared
print(r_squared)

0.796102863798257


In [29]:
# 회귀계수가 가장 큰 변수와 값 구하기
coefs = model.params
max_coef = coefs.idxmax(), coefs.max()  #튜플 타입
print(max_coef)

('engineSize', 13051.486542147013)


In [30]:
# p_value가 가장 낮은 변수와 값 구하기
pvalues = model.pvalues

min_pvalue = pvalues.idxmin(), pvalues.min()
print(min_pvalue)

('engineSize', 0.0)
